scm/driver/gitlab/webhook.go

```diff
@@ -420,6 +430,64 @@ func parseTimeString(timeString string) time.Time {
 	return t
 }
 
+func convertPipelineHook(src *pipelineHook) *scm.PipelineHook {
+	namespace, name := scm.Split(src.Project.PathWithNamespace)
+	return &scm.PipelineHook{
+		Repo: scm.Repository{
+			ID:        strconv.Itoa(src.Project.ID),
+			Namespace: namespace,
+			Name:      name,
+			Clone:     src.Project.GitHTTPURL,
+			CloneSSH:  src.Project.GitSSHURL,
+			Link:      src.Project.WebURL,
+			Branch:    src.Project.DefaultBranch,
+			Private:   false, // TODO how do we correctly set Private vs Public?
+		},
+		Commit: scm.Commit{
+			Sha:     src.ObjectAttributes.SHA,
+			Message: src.Commit.Message,
+			Author: scm.Signature{
+				Login:  src.User.Username,
+				Name:   src.Commit.Author.Name,
+				Email:  src.Commit.Author.Email,
+				Avatar: src.User.AvatarURL,
+			},
+			Committer: scm.Signature{
+				Login:  src.User.Username,
+				Name:   src.User.Name,
+				Email:  src.User.Email,
+				Avatar: src.User.AvatarURL,
+			},
+			Link: src.Commit.URL,
+		},
+		Execution: scm.Execution{
+			ID:      strconv.Itoa(src.ObjectAttributes.ID),
+			Status:  src.ObjectAttributes.Status,
+			Created: parseTimeString(src.ObjectAttributes.CreatedAt),
+			URL:     src.ObjectAttributes.URL,
+		},
+		Sender: scm.User{
+			Login:  src.User.Username,
+			Name:   src.User.Name,
+			Email:  src.User.Email,
+			Avatar: src.User.AvatarURL,
+		},
+		PullRequest: scm.PullRequest{
+			Number:  src.MergeRequest.ID,
+			Title:   src.MergeRequest.Title,
+			Sha:     src.ObjectAttributes.SHA,
+			Ref:     src.ObjectAttributes.Ref,
+			Source:  src.MergeRequest.SourceBranch,
+			Target:  src.MergeRequest.TargetBranch,
+			Link:    src.MergeRequest.URL,
+			Created: parseTimeString(src.ObjectAttributes.CreatedAt),
+			Draft:   false,
+			Closed:  false,
+			Merged:  false,
+		},
```

Comment:
These values aren't coming from gitlab, so we can ignore them.

Reply:
Done